In [1]:
import numpy as np
import pandas as pd
import matplotlib.pyplot as plt
import seaborn as sns

<h1> IMPORTING PENGUIN DATA SET USING PANDAS</h1>

In [2]:
#NAMING DATASET

penguins = pd.read_csv('dataset\\penguins.csv')

In [3]:
#CALLING DATASET

penguins

species     island  bill_length_mm  bill_depth_mm  flipper_length_mm  \
0       Adelie  Torgersen            39.1           18.7              181.0   
1       Adelie  Torgersen            39.5           17.4              186.0   
2       Adelie  Torgersen            40.3           18.0              195.0   
3       Adelie  Torgersen             NaN            NaN                NaN   
4       Adelie  Torgersen            36.7           19.3              193.0   
..         ...        ...             ...            ...                ...   
339  Chinstrap      Dream            55.8           19.8              207.0   
340  Chinstrap      Dream            43.5           18.1              202.0   
341  Chinstrap      Dream            49.6           18.2              193.0   
342  Chinstrap      Dream            50.8           19.0              210.0   
343  Chinstrap      Dream            50.2           18.7              198.0   

     body_mass_g     sex  year  
0         3750.0    male  2007  
1         3800.0  female  2007  
2         3250.0  female  2007  
3            NaN     NaN  2007  
4         3450.0  female  2007  
..           ...     ...   ...  
339       4000.0    male  2009  
340       3400.0  female  2009  
341       3775.0    male  2009  
342       4100.0    male  2009  
343       3775.0  female  2009  

[344 rows x 8 columns]

<h1>BASIC CHECKING AND DESCRIPTIVE ANALYTICS

In [4]:
penguins.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 344 entries, 0 to 343
Data columns (total 8 columns):
 #   Column             Non-Null Count  Dtype  
---  ------             --------------  -----  
 0   species            344 non-null    object 
 1   island             344 non-null    object 
 2   bill_length_mm     342 non-null    float64
 3   bill_depth_mm      342 non-null    float64
 4   flipper_length_mm  342 non-null    float64
 5   body_mass_g        342 non-null    float64
 6   sex                333 non-null    object 
 7   year               344 non-null    int64  
dtypes: float64(4), int64(1), object(3)
memory usage: 21.6+ KB


In [5]:
penguins.describe()

bill_length_mm  bill_depth_mm  flipper_length_mm  body_mass_g  \
count      342.000000     342.000000         342.000000   342.000000   
mean        43.921930      17.151170         200.915205  4201.754386   
std          5.459584       1.974793          14.061714   801.954536   
min         32.100000      13.100000         172.000000  2700.000000   
25%         39.225000      15.600000         190.000000  3550.000000   
50%         44.450000      17.300000         197.000000  4050.000000   
75%         48.500000      18.700000         213.000000  4750.000000   
max         59.600000      21.500000         231.000000  6300.000000   

              year  
count   344.000000  
mean   2008.029070  
std       0.818356  
min    2007.000000  
25%    2007.000000  
50%    2008.000000  
75%    2009.000000  
max    2009.000000

<h1>CREATING OWN COLUMN

In [6]:
penguins['weight_kg'] = np.round(penguins['body_mass_g']/ 100, 1)

In [19]:
penguins['weight_kg']

0      37.5
1      38.0
2      32.5
3       NaN
4      34.5
       ... 
339    40.0
340    34.0
341    37.8
342    41.0
343    37.8
Name: weight_kg, Length: 344, dtype: float64

<h1>DATA EXPLORATION</h1>

In [7]:
penguins['species'].value_counts()

species
Adelie       152
Gentoo       124
Chinstrap     68
Name: count, dtype: int64

In [8]:
penguins['island'].value_counts()

island
Biscoe       168
Dream        124
Torgersen     52
Name: count, dtype: int64

<h1>DOES THE ADELIE PENGUIN LIVE EXCLUSIVELY ON AND ISLAND

In [9]:
penguins['species'] == 'Adelie'

0       True
1       True
2       True
3       True
4       True
       ...  
339    False
340    False
341    False
342    False
343    False
Name: species, Length: 344, dtype: bool

In [10]:
penguins[penguins['species'] == 'Adelie']
penguins[penguins['island'] == 'Biscoe']

species  island  bill_length_mm  bill_depth_mm  flipper_length_mm  \
20   Adelie  Biscoe            37.8           18.3              174.0   
21   Adelie  Biscoe            37.7           18.7              180.0   
22   Adelie  Biscoe            35.9           19.2              189.0   
23   Adelie  Biscoe            38.2           18.1              185.0   
24   Adelie  Biscoe            38.8           17.2              180.0   
..      ...     ...             ...            ...                ...   
271  Gentoo  Biscoe             NaN            NaN                NaN   
272  Gentoo  Biscoe            46.8           14.3              215.0   
273  Gentoo  Biscoe            50.4           15.7              222.0   
274  Gentoo  Biscoe            45.2           14.8              212.0   
275  Gentoo  Biscoe            49.9           16.1              213.0   

     body_mass_g     sex  year  weight_kg  
20        3400.0  female  2007       34.0  
21        3600.0    male  2007       36.0  
22        3800.0  female  2007       38.0  
23        3950.0    male  2007       39.5  
24        3800.0    male  2007       38.0  
..           ...     ...   ...        ...  
271          NaN     NaN  2009        NaN  
272       4850.0  female  2009       48.5  
273       5750.0    male  2009       57.5  
274       5200.0  female  2009       52.0  
275       5400.0    male  2009       54.0  

[168 rows x 9 columns]

<h1>ADELIE,GENTOO, and, CHINSTRAP in all ISLAND

penguins[(penguins['species'] == 'Adelie') & (penguins['island'] == 'Biscoe')].info()
penguins[(penguins['species'] == 'Adelie') & (penguins['island'] == 'Dream')].info()
penguins[(penguins['species'] == 'Adelie') & (penguins['island'] == 'Torgersen')].info()

In [12]:
penguins[(penguins['species'] == 'Gentoo') & (penguins['island'] == 'Biscoe')].info()
penguins[(penguins['species'] == 'Gentoo') & (penguins['island'] == 'Dream')].info()
penguins[(penguins['species'] == 'Gentoo') & (penguins['island'] == 'Torgersen')].info()

<class 'pandas.core.frame.DataFrame'>
Index: 124 entries, 152 to 275
Data columns (total 9 columns):
 #   Column             Non-Null Count  Dtype  
---  ------             --------------  -----  
 0   species            124 non-null    object 
 1   island             124 non-null    object 
 2   bill_length_mm     123 non-null    float64
 3   bill_depth_mm      123 non-null    float64
 4   flipper_length_mm  123 non-null    float64
 5   body_mass_g        123 non-null    float64
 6   sex                119 non-null    object 
 7   year               124 non-null    int64  
 8   weight_kg          123 non-null    float64
dtypes: float64(5), int64(1), object(3)
memory usage: 9.7+ KB
<class 'pandas.core.frame.DataFrame'>
Index: 0 entries
Data columns (total 9 columns):
 #   Column             Non-Null Count  Dtype  
---  ------             --------------  -----  
 0   species            0 non-null      object 
 1   island             0 non-null      object 
 2   bill_length_mm     0 non-

In [13]:
penguins[(penguins['species'] == 'Chinstrap') & (penguins['island'] == 'Biscoe')].info()
penguins[(penguins['species'] == 'Chinstrap') & (penguins['island'] == 'Dream')].info()
penguins[(penguins['species'] == 'Chinstrap') & (penguins['island'] == 'Torgersen')].info()

<class 'pandas.core.frame.DataFrame'>
Index: 0 entries
Data columns (total 9 columns):
 #   Column             Non-Null Count  Dtype  
---  ------             --------------  -----  
 0   species            0 non-null      object 
 1   island             0 non-null      object 
 2   bill_length_mm     0 non-null      float64
 3   bill_depth_mm      0 non-null      float64
 4   flipper_length_mm  0 non-null      float64
 5   body_mass_g        0 non-null      float64
 6   sex                0 non-null      object 
 7   year               0 non-null      int64  
 8   weight_kg          0 non-null      float64
dtypes: float64(5), int64(1), object(3)
memory usage: 0.0+ bytes
<class 'pandas.core.frame.DataFrame'>
Index: 68 entries, 276 to 343
Data columns (total 9 columns):
 #   Column             Non-Null Count  Dtype  
---  ------             --------------  -----  
 0   species            68 non-null     object 
 1   island             68 non-null     object 
 2   bill_length_mm     68 n

<h1>Gentoo Species

In [14]:
# penguins[penguins['island'] == 'Dream'].count()

penguins[(penguins['species'] == 'Gentoo') & (penguins ['island'] == 'Torgersen')].count()



species              0
island               0
bill_length_mm       0
bill_depth_mm        0
flipper_length_mm    0
body_mass_g          0
sex                  0
year                 0
weight_kg            0
dtype: int64

In [15]:
gentoo_df = penguins[(penguins['species'] == 'Gentoo') & (penguins ['island'] == 'Biscoe')].count()
gentoo_df['weight_kg'].mean()

np.float64(123.0)

In [16]:
np.round(gentoo_df['weight_kg'].mean(), 1)

np.float64(123.0)

In [17]:
penguins[(penguins['species'] == 'Adelie') & (penguins ['island'] == 'Torgersen')].count()

species              52
island               52
bill_length_mm       51
bill_depth_mm        51
flipper_length_mm    51
body_mass_g          51
sex                  47
year                 52
weight_kg            51
dtype: int64

In [20]:
penguins[(penguins['species'] == 'Adelie') & (penguins ['island'] == 'Torgersen')].count()

species              52
island               52
bill_length_mm       51
bill_depth_mm        51
flipper_length_mm    51
body_mass_g          51
sex                  47
year                 52
weight_kg            51
dtype: int64

<h1>Insight 1</h1>
<h2>Residence of Adelie in every Island</h2>

In [22]:
adel = penguins[penguins['species'] == 'Adelie']
adel['island'].value_counts()

island
Dream        56
Torgersen    52
Biscoe       44
Name: count, dtype: int64

<h1>Their is 56 Adelie in Dream, 52 in Trogersen and 44 in Biscoe</h1>

<h1>Insight 2</h1>
<h2>Residence of Chinstrap in every Island</h2>

In [25]:
chins = penguins[penguins['species'] == 'Chinstrap']
chins['island'].value_counts()

island
Dream    68
Name: count, dtype: int64

<h1>Their is 68 Chinstrap in Dream, 0 in Trogersen and 0 in Biscoe<h1>

<h1>Insight 3</h1>
<h2>Residence of Gentoo in every Island</h2>

In [28]:
gent = penguins[penguins['species'] == 'Gentoo']
gent['island'].value_counts()

island
Biscoe    124
Name: count, dtype: int64

<h1>Their is 0 Gentoo in Dream, 0 in Trogersen and 124 in Biscoe<h1>

<h1>Insight 4</h1>
<h1>Weight per species</h1>

In [ ]:
penguin_biscoe = penguins[(penguins['species'] == 'Adelie') & (penguins['island'] == 'Biscoe')]
p